tests/src/test/scala/pureconfig/ConfigCursorSuite.scala

```diff
@@ -37,6 +37,72 @@ class ConfigCursorSuite extends BaseSuite {
       WrongType(ConfigValueType.OBJECT, Set(ConfigValueType.STRING)), defaultPathStr)
   }
 
+  it should "allow being cast to boolean in a safe way" in {
+    cursor("true").asBoolean shouldBe Right(true)
+    cursor("false").asBoolean shouldBe Right(false)
+
+    cursor("abc").asBoolean should failWith(
+      WrongType(ConfigValueType.STRING, Set(ConfigValueType.BOOLEAN)))
+    cursor("1").asBoolean should failWith(
```

Comment:
@mbeatrizmagalhaes The unit test is [failing](https://travis-ci.org/pureconfig/pureconfig/jobs/449084988) here because it's looking for `ConfigValueType.NUMBER` for `"1"`.